tools/proto2jsonschema/postgen.js

```diff
@@ -0,0 +1,46 @@
+const path = require('path');
+const fs = require('fs');
+const recursive = require("recursive-readdir");
+const camelcaseKeys = require('camelcase-keys');
+
+/**
+ * This tool polishes the JSON schemas with a few modifications:
+ * - Adds "$id" – "." delimited ID – e.g. "google.events.cloud.audit.v1.LogEntryData"
+ * - Adds "name" – The name for the JSON schema – e.g. "LogEntryData"
+ * - Removes snake_case fields that are duplicative of camelCase fields that are seen in JSON payloads
+ *   - This is a feature request for the proto2jsonschema generator.
+ */
+const ROOT = path.resolve(`${__dirname}/../../jsonschema`);
+console.log(`Fixing paths in dir: ${ROOT}`);
+(async () => {
+  const filePaths = await recursive(ROOT);
+  // For every file
+  filePaths.map(filePath => {
+    const dataName = path.basename(filePath,  path.extname(filePath)); // i.e. LogEntryData
+    
+    // Create the modified JSON schema output
+    const json = JSON.parse(fs.readFileSync(filePath).toString());
```

Comment:
Might be worth a try/catch on this to point out the protoc step had a problem.

Reply:
Good comment, however I think JSON syntax validation after the protoc step should be separate.

If there are FS errors, I think reading those FS errors rather than any caught errors would be easier to debug.